zipkin-common/src/main/scala/com/twitter/zipkin/common/Trace.scala

```diff
@@ -0,0 +1,11 @@
+package com.twitter.zipkin.common
+
+/** Utilities for working on traces */
+object Trace {
+
+  /** What's the total duration of the spans in this trace? */
+  def duration(spans: List[Span]): Long = {
+    val endTs = spans.flatMap(_.annotations).map(_.timestamp).reduceOption(_ max _)
+    (endTs.getOrElse(0L) - spans.headOption.flatMap(_.timestamp).getOrElse(0L))
```

Comment:
span.duration I mean.. anyway I agree it is weird and will move the logic from that branch here.